url088e9d87be_jpush_jpush_java_library_tgz-pJ8-cn_jpush_example_JPushClientExampleJ8/src/cn/jpush/api/HttpPostClient.java

```diff
@@ -0,0 +1,146 @@
+package cn.jpush.api;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.TrustManager;
+
+public class HttpPostClient {
+
+	private final String CHARSET = "UTF-8";
+
+	//设置连接超时时间
+	private final int DEFAULT_CONNECTION_TIMEOUT = (20 * 1000); // milliseconds
+	//设置读取超时时间
+	private final int DEFAULT_SOCKET_TIMEOUT = (30 * 1000); // milliseconds
+
+	public MessageResult post(final String path, final boolean enableSSL, final MessageParams messageParams) {
+		MessageResult messageResult = ValidateRequestParams.vidateParams(messageParams);
+		if(messageResult != null) return messageResult;
+		
+		byte[] data = null;
+		try {			
+			data = parse(messageParams).getBytes(CHARSET);
+		} catch (UnsupportedEncodingException e) {
+			e.printStackTrace();
+		}
+		return sendPost(path, enableSSL, data);
+	}
+
+	protected MessageResult sendPost(final String path, final boolean enableSSL, byte[] data){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.